### ISSUE TITLE
Unpacking Addon of 30MB+ sizes causes client freeze and then non-responsive behaviour

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#7227]
### Game and System Information

- **Version:** 1.17.11+dev (73a1b4)
- **Downloaded from:** Github
- **Build info:** Msys2
- **OS:** Windows 10

### Description of the bug

Downloading addons of medium to large size works but unpacking the addon causes the game to freeze and then become non-responsive.

### Steps to reproduce the behavior

1. Addons server
2. Download BIG addon
3. get "Installing Add-on: <Title>"
4. Click screen a few times after 5 secs

### Expected behavior

No freeze and no loss of response

### Additional context

![image](https://user-images.githubusercontent.com/5283677/209392971-2d04058c-ed99-44d2-a0f3-6df8d64d6daf.png)

**Comment by Pentarctagon** [comment]
Does it eventually unfreeze?

**Comment by knyghtmare** [comment]
> Does it eventually unfreeze?

Nah, it crashes to desktop.

**Comment by Pentarctagon** [comment]
Can you post the logs from a time when it crashed? On Linux I do get a popup about wesnoth not responding, but it does eventually finish installing successfully.

**Comment by Toranks** [comment]
On Windows 7 this happens:

- I update Eastern Invasion Revised > Crashed to desktop immediately
- Restart and go to Addons > EIR isn't installed
- Install EIR again from scratch > Download correctly, and when the download ends, this error happens:
![imagen](https://user-images.githubusercontent.com/40789364/209394940-52e76b27-32fb-4e44-a6c4-2f3c3db23991.png)
(Translation: A problem occurred while trying to create the files needed to install this plugin )
- Go to addons folder, and try to browse inside > This error happens:
![imagen](https://user-images.githubusercontent.com/40789364/209395089-967889a2-a6a5-4cc6-9210-19ceffc2942d.png)
(Translation: Can't get access to -any mod folder-. Access denied.
- Delete the entire addon folder
- Start the game and go to addons server again
- Install EIR
- Profit! Finally works.

**Comment by Toranks** [comment]
CRASH
[wesnoth-20221223-193125-1061173414.log](https://github.com/wesnoth/wesnoth/files/10296983/wesnoth-20221223-193125-1061173414.log)

INSTALLATION NOT PERMITTED
[wesnoth-20221223-195829-2233947319.log](https://github.com/wesnoth/wesnoth/files/10296982/wesnoth-20221223-195829-2233947319.log)

**Comment by Toranks** [comment]
"C:\Users\USER\Documents\My Games\Wesnoth1.17/data/add-ons/Eastern_Invasion_Dalas\lua"

These slashes mixing / and \ could be the cause of the error?

**Comment by Pentarctagon** [comment]
Access Denied errors mean that for some reason Wesnoth isn't allowed to access that folder. Usually that seems to happen due to antivirus software restricting what's able to access certain folders.

**Comment by loonycyborg** [comment]
It also could be caused by lack of hdd space. If it happens during install it might be useful to monitor cpu and disk i/o just before crash.

**Comment by Toranks** [comment]
There is a possibility that in my case it happened because I altered the content of the addon folder between the previous update and this one. (In this case, I added the spanish translation)

**Comment by Wedge009** [comment]
This issue occurs in self-compiled build? I just tried 1.16 release for Linux, Windows 10, and Windows 7. There's no Eastern Invasion Revised in 1.16 so I just used Napoleonic Wars, over 80 GiB download. In Windows 10, I noticed the 'not responding' feedback in the (presumably) unpacking stage at the end but it was rather brief, no more than a second. For Windows 7 and Linux releases, I didn't notice any such unresponsiveness.

I also tried Linux master build (17a95134acd) with Eastern Invasion Revised and noticed a new progress dialogue after the download progress, but it was over so quickly I didn't see what it was - I assume it was the unpacking progress.

I don't think the actual size is a factor here - I note Eastern Invasion Revised appears to be a 14 MiB download, unless the 30 in the title is intended to refer to the unpacked size. I also observed the unresponsive behaviour in Windows 10 just downloading the add-ons list for 1.16 and that's only about 1.5 MiB.

I would suggest seeing if any of these issues actually occur in the official builds. I've had [problems with download operations in Windows builds from MSVC](https://github.com/wesnoth/wesnoth/issues/6171), for example.
